extjsdk/README.md

```diff
@@ -0,0 +1,93 @@
+
+#### Authentication
+The authentication handler receives all messages until and including the message that marks the authentication as successful. A response to a successful authentication has a status code of 200 and includes a mass of information about the connection and its privileges inside the message body, none of which is necessary to know for the purposes of connecting to the source. A response to a failed authentication will have a status code of 300+, typically 400, and the body may contain a message specifying why the authentication failed. Every Client has a default authentication handler that logs whether the authentication succeeded or failed. The authentication handler receives Response objects.
+
+#### HTTP
+The HTTP handler receives all non source-related messages after authentication has succeeded. The majority of these messages will be confirmations of receipt for a message sent to the Vantiq server, and consist of a status of 200 and little else. Any other messages are likely to be an error of some sort, with status code 300+ and a body containing a message describing the error. The HTTP handler receives Response objects.
+
+#### <a name="closeHandler" id="closeHandler">Closure
+The closure handler does not deal with a specific message or type of message, but instead is called when either your code calls `client.close()` or the WebSocket connection is forced to close, most likely due to a problem with the connection. This handler is called after everything but the handlers are reset, essentially creating a new Client targeting the same source. If you have saved a reference to the Client's Listener, be aware that the Listener is stopped and replaced with a functionally identical Listener just before this handler is called. The closure handler receives the Client whose connection closed.
+
+#### Configuration
+The configuration handler receives the message containing a configuration document created on the Vantiq client. This message is only received upon a successful source connection, and contains the source's configuration document as a Map at `((Map) <message>.getObject()).get("config")`. The configuration handler receives an ExtensionServiceMessage.
+
+#### Reconnection
+Under certain circumstances, typically a change to the source's configuration document, the Vantiq server may disconnect from the source. The reconnection handler is called when this occurs. It is suggested that in this handler you reset any changes caused by the configuration document and then reconnect using `client.connectToSource()`. If you do not want to send the reconnect message yourself or do not wish to create your own handler, then calling `client.setAutoReconnect(true)` will tell the Client to automatically try to reconnect once any handlers are finished The reconnection handler receives an ExtensionServiceMessage.
```

Comment:
It was a thought that occurred because in my UDP source I set/change several variables based on the configuration document, and if the port that it binds to were to change, for example, then some of the old settings would stick around and cause problems if they weren't removed here. I probably should specify the original config though.

Reply:
Yeah.  Maybe there's some other section that should describe the overall flow (which can be added in a separate PR).

For the source I'm working on, I expect the config document to contain all that.  So the only local configuration will likely be "how to get to VANTIQ".  Whether that's the right way or only way, I'm not certain.  But that's the way I'm thinking about it -- the specific source's actions are dictated by the configuration document.